odoo/tools/date_utils.py

```diff
@@ -199,6 +199,35 @@ def subtract(value, *args, **kwargs):
     """
     return value - relativedelta(*args, **kwargs)
 
+def ceil(value, granularity):
+    """
+    Return the ``value`` rounded up to a ``granularity``.
+
+    :param value: initial date or datetime.
+    :param granularity: year, month, day, hour, minute or second
+    :return: the resulting date/datetime.
+    """
+    granularities = (
+        ['year', 'month', 'day', 'hour', 'minute', 'second', 'microsecond']
+        if isinstance(value, datetime) else
+        ['year', 'month', 'day']
+    )
+    index = granularities.index(granularity, 0, len(granularities) - 1)
```

Comment:
Also `-1` works fine as an end index.

And I wouldn't use start/end bounds. If we don't want to support rounding to the smallest unit, then it should probably be checked explicitely with a clear error message e.g.
```suggestion
    assert granularity != granularities[-1], f"rounding to the smallest possible granularity ({granularity!r}) is not useful"
    index = granularities.index(granularity)
```
But I'm not sure that's very useful, considering the rest fo the code it doesn't seem like it even avoids an unnecessary increment? There would be no fields following the specified granularity so the `carry` would be `False`, therefore the value wouldn't be incremented. Seems fine.

Reply:
Also might be worth catching the `ValueError` and providing a more helpful error message: this would just raise e.g. "days is not in the list" but you may have no idea what *is*.